projects/js-packages/charts/src/providers/chart-context/chart-context.tsx

```diff
@@ -1,22 +1,25 @@
-import { createContext, useContext, useCallback, useRef, useMemo } from 'react';
+import { createContext, useContext, useCallback, useRef, useState, useMemo } from 'react';
 import type { ChartContextValue, ChartRegistration } from './types';
 import type { FC, ReactNode } from 'react';
 
-const ChartContext = createContext< ChartContextValue | null >( null );
+export const ChartContext = createContext< ChartContextValue | null >( null );
 
 export interface ChartProviderProps {
 	children: ReactNode;
 }
 
 export const ChartProvider: FC< ChartProviderProps > = ( { children } ) => {
 	const chartsRef = useRef< Map< string, ChartRegistration > >( new Map() );
+	const [ version, setVersion ] = useState( 0 );
```

Comment:
What is the problem `version` is solving here?

Reply:
Good question 🙈 Thank you! 
The version counter was a workaround for using useRef when the chart registry actually needs to trigger re-renders for Legend components. I've updated it to use useState directly with immutable Map updates instead, which is much cleaner since useState handles re-rendering automatically.